### DISCUSSION TITLE
8.x releases and backports

### DISCUSSION THREAD

**Discussion by oliverklee** [MyIntervals/PHP-CSS-Parser#615] [General]
Hi @sabberworm and @JakeQZ,

now that I've just merged our first to-be-backported feature #607, I propose this approach:

1. backport the refactoring #599
2. release version 8.5.2 and target the 8.x branch for 8.6.0
3. backport the feature #607

What do you think?

**Comment by oliverklee** [comment]
And @JakeQZ, would you be willing to do the backports? (I'd be willing to take care of the release.)

**Reply by JakeQZ** [reply]
> And @JakeQZ, would you be willing to do the backports? (I'd be willing to take care of the release.)

Seems fair enough :)

**Comment by JakeQZ** [comment]
I added the 8.6 milestone since it became clear that there were a couple of urgent features to be put in.

> 1. backport the refactoring

This isn't necessary.  The refactoring seems to be completely indepentent of the other changes from the original PR (#390).

> 2. release version 8.5.2 and target the 8.x branch for 8.6.0

I'm not hearing any cries for the bugfixes.  But am hearing cries for the missing features.  I suggest just retarget 8.x banch for 8.6.0, and skip 8.5.2.

> 3. backport the feature #607

Yes, this is being asked for.

**Reply by oliverklee** [reply]
> I'm not hearing any cries for the bugfixes. But am hearing cries for the missing features. I suggest just retarget 8.x banch for 8.6.0, and skip 8.5.2.

I'd still like to release 8.5.2 to ship the bug fixes and `@internal` markers to people quickly. Apart from a few minutes of work, I see no downside to that. Would you be okay with that? (For you, that would only mean reviewing and merging the release PR.)

**Reply by JakeQZ** [reply]
I'd be OK with that.  I just didn't see much point now for 8.5.2, given that we'd be ready for 8.6.0 fairly soon.  (And then possibly 8.7.0, with the enhancements that are not so straightforward, depending on how well we progress towards 9.0.)

**Comment by oliverklee** [comment]
Closing, as we've done those 8.x releases in the meantime.
